src/test/debuginfo/thread-names.rs

```diff
@@ -0,0 +1,29 @@
+// compile-flags:-g
+
+// === GDB TESTS ===================================================================================
+//
+// gdb-command:run
```

Comment:
Also add lldb tests? I feel like the mac CI only runs those so it'd be good to make sure it keeps working there too.